src/Tickets/Commerce/Gateways/Stripe/Webhooks/Handler.php

```diff
@@ -129,6 +129,31 @@ public static function get_handler_method_for_event( $type ) {
 	 * @return bool|WP_Error|null
 	 */
 	public static function update_order_status( \WP_Post $order, Commerce_Status\Status_Interface $status, array $metadata = [] ) {
+		if ( tribe( Order::class )->is_order_locked( $order->ID ) || ! tribe( Order::class )->is_checkout_completed( $order->ID ) ) {
+
+			if ( function_exists( 'as_enqueue_async_action' ) ) {
```

Comment:
This method call is already happening in the context of handling a request from the Payment Gateway: it's already a background (i.e. not user-triggered) request, why queue an action and add further delay and complexity?